src/ripple/conditions/impl/condition.cpp

```diff
@@ -0,0 +1,167 @@
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/contract.h>
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/hashlock.h>
+#include <ripple/conditions/impl/utils.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <vector>
+#include <iostream>
+
+namespace ripple {
+namespace cryptoconditions {
+
+boost::optional<condition_t>
+load_condition(std::string s)
+{
+    static boost::regex const re_current (
+            "^"                            // start of line
+            "cc:"                          // 'cc' for cryptocondition
+            "([1-9a-f][0-9a-f]{0,3}|0):"   // type (hexadecimal)
+            "([1-9a-f][0-9a-f]{0,15}):"    // feature bitmask (hexadecimal)
+            "([a-zA-Z0-9_-]{0,86}):"       // fingerprint (base64url)
+            "([1-9][0-9]{0,17}|0)"         // fulfillment length (decimal)
+            "$"                            // end of line
+        , boost::regex_constants::optimize
+    );
+
+    boost::smatch match;
+
+    if (!boost::regex_match (s, match, re_current))
+        return boost::none;
+
+    try
+    {
+        condition_t c;
+
+        c.type = parse_hexadecimal<std::uint16_t> (match[1]);
+
+        if (!isCondition (c.type))
+            return boost::none;
+
+        c.requires = parse_hexadecimal<std::uint32_t>(match[2]);
+        c.fulfillment_length = parse_decimal<std::uint16_t>(match[4]);
+
+        if (c.fulfillment_length > maxFulfillmentLength)
+            return boost::none;
+
+        // TODO: Avoid copying by decoding directly
+        // into the condition's buffer
+        auto fingerprint = base64url_decode(match[3]);
+
+        if (fingerprint.size() != c.fingerprint.size())
+            return boost::none;
+
+        std::memcpy(
+            c.fingerprint.data(),
+            fingerprint.data(),
+            fingerprint.size());
+
+        return c;
+    }
+    catch (std::exception const& x)
+    {
+        return boost::none;
+    }
+}
+
+boost::optional<condition_t>
+load_condition(Slice const& s)
+{
+    try
+    {
+        auto iter = s.cbegin();
+
+        condition_t c;
+
+        std::tie (iter, c.type) =
+            oer::decode_integer<std::uint16_t> (
+                iter, s.cend());
+
+        if (!isCondition (c.type))
+            return boost::none;
+
+        std::tie (iter, c.requires) =
+            oer::decode_varuint<std::uint32_t> (
+                iter, s.cend());
+
+        iter = oer::decode_octetstring (
+            iter, s.cend(),
+            c.fingerprint.data(),
+            c.fingerprint.size());
```

Comment:
If the fingerprint sent is shorter than `c.fingerprint.size()` then we don't detect the error and some of `fingerprint` will have garbage.